Common > Validation: 유효성 응답 모델에 "array" 타입 프로퍼티 모델 추가

merge-simpson opened the pull request:
<!--
<br /><br /><br /><br /><br />

<table align="center" border="30">
<tr>
  <td>🏗️ 선행 작업 병합 후, 리베이스할 예정입니다.</td>
</tr>
</table>

<br /><br /><br /><br /><br />
그니까 아직 리뷰하지 말라니깐용.
<br /><br /><br /><br /><br />
<br /><br /><br /><br /><br />
-->

## Issues

- Resolves #331 

<br />

## Description

- Array(list) 데이터의 유효성 모델을 추가합니다.
- Rel: #323 

<br />

## Review Points

- 노션 스펙을 참고하실 수 있습니다. (로그인 필요)  
  https://www.notion.so/nettee/Auth-User-API-25001973b23a800c9e9fe2392aa7fa97#25001973b23a802aad45ea139d9f1743
- 관련 파일을 참고하실 수 있습니다.  
  - common/src/main/java/nettee/common/validation/model/interfaces/**BaseValidationProperty**.java  
    ```java
    public interface BaseValidationProperty {
        String type();
        Boolean required();
        Map<String, String> messages();
        
        // ... (중략)
    }
    ```  
  - common/src/main/java/nettee/common/validation/model/**ValidationResponseModel**.java

<br />

## How Has This Been Tested?

- 테스트 생략

<br />

## Additional Notes

### 다이어그램

```mermaid
classDiagram
direction TB

%% ================= Base & Common =================
class BaseValidationProperty {
  <<interface>>
  +type() String
  +required() Boolean
  +messages() Map~String,String~
}

class LengthValidationProperty {
  <<interface>>
  +minLength() Integer
  +maxLength() Integer
}
BaseValidationProperty <-- LengthValidationProperty

class ArrayValidationProperty {
  <<record>>
  +type() String
  +required() Boolean
  +minLength() Integer
  +maxLength() Integer
  +messages() Map~String,String~
}
LengthValidationProperty <.. ArrayValidationProperty : implements
```

merge-simpson committed:
feat(common): implement array validation property model

Add `ArrayValidationProperty` implementing `LengthValidationProperty` with `type`, `required`, `minLength`, `maxLength`, and `messages`.
Includes validation for type consistency, length bounds, and default initialization for required and messages.
Auto-populates messages for `required`, `minLength`, and `maxLength` fields and ensures immutability.

merge-simpson commented:
- Resolves #323

wowddok99 commented on common/src/main/java/nettee/common/validation/model/ArrayValidationProperty.java:
```diff
@@ -0,0 +1,60 @@
+        Map<String, String> messages
+) implements LengthValidationProperty {
+    public ArrayValidationProperty {
+        assert type == null || type.isBlank() || "string".equals(type)
```
```java
assert type == null || type.isBlank() || "array".equals(type)
        : "type must be 'array'. If new spec types are added, update this assertion accordingly.\n" +
        "type 속성은 반드시 'array'여야 합니다. 만약 새로운 값을 추가한다면 이 assert 구문을 함께 고쳐야 합니다.";
```
- 현재 assert 메시지가 string을 요구하고 있는데 PR 제목 및 노션 스펙상으로 array가 맞을 것 같습니다. 아마 이전에 작업하시던 내용을 붙여넣으면서 생긴 작은 실수같습니다. 위와 같이 수정 요청 드립니다!!

merge-simpson replied:
발견해 주신 부분 수정하여 바로 게시했습니다! 병합하겠습니다.
섬세한 검토 감사합니다!!!

wowddok99 reviewed (approved):
코드 잘 확인했습니다! 
몇 가지 확인이 필요한 부분에 코멘트 남겨두었으니, 시간 되실 때 한번 봐주시면 감사하겠습니다. 
해당 부분들 수정 완료되면 바로 머지해주시면 됩니다. 
작업하시느라 고생 많으셨습니다! (Approve는 미리 해두겠습니다 👍)

merge-simpson committed:
fix(common): correct default `type` for `ArrayValidationProperty`

Changed assertion and default assignment from `"string"` to `"array"` to properly reflect the property’s expected type.
